packages/next-codemod/bin/upgrade.ts

```diff
@@ -172,12 +172,33 @@ export async function runUpgrade(
 
   fs.writeFileSync(appPackageJsonPath, JSON.stringify(appPackageJson, null, 2))
 
-  const nextDependency = `next@${targetNextVersion}`
-  const dependenciesToInstall = [
-    `react@${targetReactVersion}`,
-    `react-dom@${targetReactVersion}`,
-  ]
+  const dependenciesToInstall = []
   const devDependenciesToInstall = []
+
+  const corePackageNameVersionMapping = {
+    react: `react@${targetReactVersion}`,
+    'react-dom': `react-dom@${targetReactVersion}`,
+    next: `next@${targetNextVersion}`,
+  }
+  for (const packageName of Object.keys(corePackageNameVersionMapping)) {
+    // First check if it's dev dependencies, then add into dev deps;
+    // If it's not, fallback into dependencies.
+    if (appPackageJson.devDependencies?.[packageName]) {
+      devDependenciesToInstall.push(corePackageNameVersionMapping[packageName])
+    } else {
+      const isPeerDep = appPackageJson.peerDependencies?.[packageName]
+      const isDep = appPackageJson.dependencies?.[packageName]
+
+      //
+      if (isPeerDep && !isDep) {
```

Comment:
We can simplify this and only update it in `dependencies` and `devDependencies`. `peerDependencies` we can ignore since that's part of the API contract of libraries and needs to be decided by library maintainers.